R/deployApp.R

```diff
@@ -512,34 +499,46 @@ getAppById <- function(id, account = NULL, server = NULL, hostUrl = NULL) {
 }
 
 applicationForTarget <- function(client, accountInfo, target, forceUpdate) {
-
-  if (is.null(target$appId)) {
-    # list the existing applications for this account and see if we
-    # need to create a new application
-    app <- getAppByName(client, accountInfo, target$appName)
-  } else {
-    # we already know the app's id, so just retrieve the rest of the metadata
-    app <- client$getApplication(target$appId)
+  # Use appId from previous deployment
+  if (!is.null(target$appId)) {
+    return(client$getApplication(target$appId))
```

Comment:
What should happen if the target app ID has been deleted? The old flow would leave `app = NULL` and try some other option.

Reply:
I just tried it out, and I think if the application id no longer exists, then `getApplication()` will error with a 404. That's probably worth considering in more detail (and I'll add a fall back in case `getApplication()` does return `NULL` in other cases), but I think it's out of scope for this PR.